File: easy-level/14. longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    if (strs.length === 0) return "";
    if (strs.some(str => str === "")) return "";
  
    var characters = strs.map(str => str.split("").shift());
  
    var isCommonPrefix = characters.every(character => character === characters[0]);
    
    var result = isCommonPrefix ? characters[0] : "";
    var nextPrefix = isCommonPrefix ? longestCommonPrefix(strs.map(str => str.substring(1))) : "";
  
    return result + nextPrefix;
  };

File: easy-level/14. longest-common-prefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        if "" in strs:
            return ""
        
        characters = list(map(lambda x: x[0], strs))
        isCommonPrefix = all(c == characters[0] for c in characters)

        result = characters[0] if isCommonPrefix else ""
        nextPrefix = self.longestCommonPrefix(list(map(lambda x: x[1:len(x)], strs))) if isCommonPrefix else ""

        return result + nextPrefix